opentelemetry-sdk/src/metrics/mod.rs

```diff
@@ -149,6 +149,61 @@ mod tests {
     // "multi_thread" tokio flavor must be used else flush won't
     // be able to make progress!
 
+    #[tokio::test(flavor = "multi_thread", worker_threads = 1)]
+    async fn invalid_instrument_config_noops() {
+        // Run this test with stdout enabled to see output.
+        // cargo test invalid_instrument_config_noops --features=testing -- --nocapture
+        let invalid_instrument_names = vec![
+            "_startWithNoneAlphabet",
+            "utf8char锈",
+            "a".repeat(256).leak(),
+            "invalid name",
+        ];
+        for name in invalid_instrument_names {
+            let test_context = TestContext::new(Temporality::Cumulative);
+            let counter = test_context.meter().u64_counter(name).init();
+            counter.add(1, &[]);
```

Comment:
nit  - how do we validate if this recording is happening on NoOp instrument?

Reply:
we don't. We check that no metrics are exported, indirectly verifying that a noop is activated.